rust/core-lib/src/editor.rs

```diff
@@ -207,13 +206,17 @@ impl<W: Write + Send + 'static> Editor<W> {
         self.view.set_old_sel_dirty();
     }
 
-    // Apply the delta to the buffer, and store the new cursor so that it gets
-    // set when commit_delta is called.
-    //
-    // Records the delta into the CRDT engine so that it can be undone. Also contains
-    // the logic for merging edits into the same undo group. At call time,
-    // self.this_edit_type should be set appropriately.
-    fn add_delta(&mut self, iv: Interval, new: Rope, new_start: usize, new_end: usize) {
+    /// Applies the delta to the text, and updates undo state.
+    ///
+    /// Records the delta into the CRDT engine so that it can be undone. Also contains
+    /// the logic for merging edits into the same undo group. At call time,
+    /// self.this_edit_type should be set appropriately.
+    ///
+    /// This method can be called multiple times. Note that it does not update
```

Comment:
"may be called multiple times" is a little unclear. Multiple times within an update cycle / rpc call?